File: Data Structures & Algorithms/house-robber/submission-2.rs

impl Solution {
    pub fn rob(mut nums: Vec<i32>) -> i32
    {
        for i in 0..nums.len()
        {
            nums[i] = std::cmp::max(
                if i < 1 {0} else { nums[i-1] },
                if i < 2 {0} else { nums[i-2] } + nums[i],
            );
        }
        return nums.pop().unwrap_or(0);
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-2.cpp

class Solution {
public:
    int lengthOfLIS(vector<int>& nums)
    {
        auto dp = vector<int>(nums.size(),1);
        auto dp_max = 0;

        for (size_t i = 0; i < nums.size(); ++i)
        {
            for (size_t j = 0; j < i; ++j)
            {
                if (nums[j] >= nums[i]) {
                    continue;
                }
                dp[i] = max(dp[i], 1 + dp[j]);
            }
            dp_max = max(dp_max, dp[i]);
        }

        return dp_max;
    }
};
